### PR TITLE
Fix linking with a social account from signup 

### PR DESCRIPTION
**Pull request by Tug**
Requires D6892

This PR fixes the redirect that wasn't working properly after you link your WordPress.com account with your google account on signup.

It also:
- removes the Facebook button which isn't used at the moment.
- updates the "Log In" link in the masterbar to use the new login page when enabled

### Testing Instructions
- Boot this branch locally or use calypso.live and navigate incognito to [/start/social](https://calypso.live/start/social?branch=fix/social-signup-2fa)
- Click on Connect with Google and select a google account which address matches the one you have for your WordPress.com account (have 2FA setup first for this account).
- Follow the social linking flow (need to enter your password and 2FA code)
- Ensure that you are redirected back to `/start/social` at the end (which should redirect you back to `/`)
- Test the resistance to errors by entering invalid inputs (email, password, 2fa codes...)
- Test that social linking, social login, logout still works (redirect) properly

### Reviews
- [x] Code
- [x] Product

### CODE REVIEW

File: client/components/signup-form/index.jsx [hunk 1]
```diff
@@ -151,6 +151,7 @@ class SignupForm extends Component {
 			page(
 				login( {
 					isNative: config.isEnabled( 'login/native-login-links' ),
+					redirectTo: this.props.getRedirectToAfterLoginUrl,
```
Comment (yurynix) [inline client/components/signup-form/index.jsx]: That variable name is wired, looks like a function.

**Comment by yurynix** [comment]
Code looks good 👍  Thanks @Tug

File: client/components/signup-form/index.jsx (hunk 1 above)
Comment (pento) [inline client/components/signup-form/index.jsx]: It's what we use everywhere else, I'm inclined to leave it.

**Comment by pento** [comment]
While testing, I ran into a similar bug with social login, so I've fixed that here (and in D6892).

The `socialConnect()` endpoint helper wasn't using the correct variable name, so wasn't sending it to the API.

Nothing else from me, code looks good, and it works as described. A final code review on the bits I
 changed would be helpful, otherwise, it's ready to ship. 🙂

**Comment by Tug** [comment]
👍

**Pull request merged**